### PR TITLE
don't do anything if hostname matches node name 

### PR DESCRIPTION
**Pull request by bixu**
this should speed up convergence a little bit

### CODE REVIEW

**Comment by meatballhat** [comment]
The automatons are angry.  I tend to agree with them.  I'd prefer some `Mixlib::Shellout` once, at the top, then guard the whole case statement with the matching hostname check.

**Comment by bixu** [comment]
I think I fixed it, although I’m having issues with getting rubocop to run on my home machine.  Let me know if the updated version is cool for you?

File: recipes/default.rb [hunk 1]
```diff
@@ -25,11 +25,16 @@
...
-case node['platform']
-when 'smartos'
-  include_recipe 'modcloth-hostname::smartos'
-when 'centos'
-  include_recipe 'modcloth-hostname::centos'
-when 'ubuntu'
-  include_recipe 'modcloth-hostname::ubuntu'
+hostname = Mixlib::ShellOut.new("hostname")
+
+unless hostname == node.name
```
Comment (meatballhat) [inline recipes/default.rb:30]: I'm pretty sure comparing a shellout instance with `node.name` will always be false.  Instead, I think you'd have to do:

``` ruby
hostname = Mixlib::ShellOut.new('hostname')
hostname.run_command

unless hostname.stdout.chomp == node.name
# ...
end
```

Or better yet, just skip shelling out altogether (sorry I didn't think of this earlier):

``` ruby
unless Socket.gethostname == node.name
# ...
end
```

**Comment by sax** [comment]
:heart:

**Comment by meatballhat** [comment]
but my hostname oh no

**Comment by bixu** [comment]
@meatballhat, do you need this!? doesn't dockernetes fix all the ops?

**Comment by meatballhat** [comment]
ya I'm really into containment

would you like to join my network on LinkedIn

**Comment by skingry** [comment]
🔼 🔼 🔼 

I don't know about the rest of you, but with the advent of Mirai, paying for infrastructure is so 2015.  I run everything I got on compromised IP cameras and Nest thermostats now.

**Comment by bixu** [comment]
img_4133

**Comment by meatballhat** [comment]
Hi @skingry!  Advent calendars are really fun!  Your infrastructure-as-code sounds like a really interesting GitHub repository.

**Pull request closed without merging**